### PR TITLE
Machine/computer boards and frames refactor

### PR DESCRIPTION
**Pull request by Core0verload**
This PR refactors the fuck out of computer frames, machine frames and boards for both.

Visible changes:
- Both computer and machine frames now reveal the name of circuit inside on examine, if any.
- You can now deconstruct any ChemMaster/CondiMaster, and use the screwdriver on it's board to turn one into another.
- AI board is moved to AI modules in circuit imprinter.

Under the hood:
- Machine frames now properly support any sheets as components, not just cable coils. I'll replace those dumbass console screens that are used as glass panels in sleeper and cryo with proper glass sheets in a separate PR.
- Boards for computers and machines are separated in path tree now. `/obj/item/weapon/circuitboard/computer` and `/obj/item/weapon/circuitboard/machine` respectively.
- Machine boards now have a proc for filling machines with their default required parts. Any machine that started with list of parts equal to default required parts now uses this proc.
- Most of the machine boards are moved to the files of their machines.
- `/obj/machinery/constructable_frame/machine_frame` is now `/obj/structure/frame/machine`. It didn't used any machine things anyway.
- `/obj/structure/computerframe` is now `/obj/structure/frame/computer`.
- Replaces legacy text typepaths in consoles with proper typepaths.
- Replaces some shitcode and dirty hacks with less shitty code and clean, shiny hacks.
- Removes a bunch of unused, legacy vars.

**This PR changes every single fucking map.** Conflicts are inevitable. Merge ASAP.

### CODE REVIEW

**Comment by Jordie0608** [comment]
Should we preemptively put a merge conflict tag on this?

**Comment by Core0verload** [comment]
Reviewing and merging it would be better. This is a pure refactor, it has no controversial or balance-affecting anything, and the only thing that players would ever notice is circuit name on frame examine. Oh, and AI board is moved to AI modules in circuit imprinter.

File: code/game/machinery/computer/buildandrepair.dm [hunk 1]
```diff
@@ -1,271 +1,394 @@
...
+				state = 3
+				icon_state = "3"
+				var/obj/item/stack/sheet/glass/G = new (loc, 2)
+				G.add_fingerprint(user)
+			if(istype(P, /obj/item/weapon/screwdriver))
+				playsound(src.loc, 'sound/items/Screwdriver.ogg', 50, 1)
+				user << "<span class='notice'>You connect the monitor.</span>"
+				var/obj/B = new src.circuit.build_path (src.loc, circuit)
+				transfer_fingerprints_to(B)
+				qdel(src)
```
Comment (phil235) [inline code/game/machinery/computer/buildandrepair.dm]: the attackby proc needs to return ..() at the end if no special interaction has been triggered.

File: code/game/machinery/computer/buildandrepair.dm [hunk 2]
```diff
@@ -1,271 +1,394 @@
...
 	name = "circuit board (Department Management Console)"
 	build_path = /obj/machinery/computer/card/minor
 	var/target_dept = 1
-//obj/item/weapon/circuitboard/shield
+	var/list/dept_list = list("General","Security","Medical","Science","Engineering")
+
+/obj/item/weapon/circuitboard/computer/card/minor/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/screwdriver))
+		target_dept = (target_dept == dept_list.len) ? 1 : (target_dept + 1)
+		user << "<span class='notice'>You set the board to \"[dept_list[target_dept]]\".</span>"
```
Comment (phil235) [inline code/game/machinery/computer/buildandrepair.dm]: else
return ..()

File: code/game/machinery/computer/buildandrepair.dm [hunk 3]
```diff
@@ -1,271 +1,394 @@
...
+	if(istype(I,/obj/item/weapon/screwdriver))
+		if(build_path == /obj/machinery/computer/rdconsole/core)
+			name = "circuit board (RD Console - Robotics)"
+			build_path = /obj/machinery/computer/rdconsole/robotics
+			user << "<span class='notice'>Access protocols successfully updated.</span>"
+		else
+			name = "circuit board (RD Console)"
+			build_path = /obj/machinery/computer/rdconsole/core
+			user << "<span class='notice'>Defaulting access protocols.</span>"
+	return
```
Comment (phil235) [inline code/game/machinery/computer/buildandrepair.dm]: else
return ..()

File: code/game/machinery/computer/buildandrepair.dm [hunk 4]
```diff
@@ -1,271 +1,394 @@
...
+		if(!emagged)
+			contraband = !contraband
+			user << "<span class='notice'>Receiver spectrum set to [contraband ? "Broad" : "Standard"].</span>"
+		else
+			user << "<span class='notice'>The spectrum chip is unresponsive.</span>"
+	if(istype(I,/obj/item/weapon/card/emag))
+		if(!emagged)
+			contraband = TRUE
+			emagged = TRUE
+			user << "<span class='notice'>You adjust [src]'s routing and receiver spectrum, unlocking special supplies and contraband.</span>"
```
Comment (phil235) [inline code/game/machinery/computer/buildandrepair.dm]: else
return ..()

File: code/game/machinery/computer/buildandrepair.dm [hunk 5]
```diff
@@ -1,271 +1,394 @@
...
 	var/possible_destinations = ""
-/obj/item/weapon/circuitboard/labor_shuttle
+
+/obj/item/weapon/circuitboard/computer/shuttle/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/device/multitool))
+		var/chosen_id = round(input(usr, "Choose an ID number (-1 for reset):", "Input an Integer", null) as num|null)
+		if(chosen_id >= 0)
+			shuttleId = chosen_id
+		else
+			shuttleId = initial(shuttleId)
```
Comment (phil235) [inline code/game/machinery/computer/buildandrepair.dm]: else
return ..()

File: code/game/machinery/constructable_frame.dm [hunk 6]
```diff
@@ -191,31 +194,35 @@
...
 						if(!user.drop_item())
 							break
+						user << "<span class='notice'>You add [P] to [src].</span>"
 						P.loc = src
 						components += P
 						req_components[I]--
-						update_req_desc()
 						return 1
 				user << "<span class='warning'>You cannot add that to the machine!</span>"
 				return 0
```
Comment (phil235) [inline code/game/machinery/constructable_frame.dm]: return ..() if no interaction was triggered.

**Comment by phil235** [comment]
Looks good except for the missing return ..() I mentionned (and most of them aren't your fault).

**Comment by Core0verload** [comment]
Wouldn't that enable unwanted interactions, such as bashing the circuits with air tank that does nothing?

**Comment by phil235** [comment]
No, attackby() for items stops at obj/item/attackby() where it handles being picked up by plant bags and stuff. Attacks are done in /obj/attackby()

**Comment by Core0verload** [comment]
So any wrong tool applied to `/obj/structure/frame` would result in large red attack message spamming up everyone around, without any actual action behind it. Not very good.

**Comment by Core0verload** [comment]
I added all `return ..()` you wanted except the ones for actual frames. I don't want to bash the frame with tools every time I accidentally use the tool in wrong step.

**Comment by KazeEspada** [comment]
But core, that would make it consistant with surgery. Where I bash the patient with the wrong tool in the wrong step.

**Comment by Core0verload** [comment]
And is bashing your patient with tools a good thing? No, it fucking sucks. We don't need consistency of suck.

**Comment by KorPhaeron** [comment]
@phil235 

Is this okay now?

**Comment by KorPhaeron** [comment]
@Core0verload 

Also maybe return attackby on any intent other than help? If @phil235 still thinks it's important

**Comment by phil235** [comment]
Nah, it's alright, but we're going to need to fix correctly the problem of structure/machines that shouldn't get hit when clicked on intent other than harm with an item sometimes in the future.

**Pull request merged**